File: Project/backend/app/models/analytics.py
from pydantic import BaseModel
from typing import List, Optional

class StatisticaMezzo(BaseModel):
    mezzo_di_spostamento: str
    conteggio: int

class StatisticaFeedback(BaseModel):
    is_utile: bool
    conteggio: int

class HeatmapPoint(BaseModel):
    lat: float
    lon: float

class StatisticaPOI(BaseModel):
    id_poi: int
    nome_poi: str
    totale_eventi: int

class DashboardResponse(BaseModel):
    statistiche_mezzi: List[StatisticaMezzo]
    statistiche_feedback: List[StatisticaFeedback]
    poi_piu_attivi: List[StatisticaPOI]

File: Project/ContextAware_app/app/src/main/java/com/example/myapplication/data/model/analytics.kt
package com.example.myapplication.data.model

import kotlinx.serialization.Serializable

@Serializable
data class StatisticaMezzo(
    val mezzo_di_spostamento: String,
    val conteggio: Int
)

@Serializable
data class StatisticaFeedback(
    val is_utile: Boolean,
    val conteggio: Int
)

@Serializable
data class HeatmapPoint(
    val lat: Double,
    val lon: Double
)

@Serializable
data class StatisticaPOI(
    val id_poi: Int,
    val nome_poi: String,
    val totale_eventi: Int
)

@Serializable
data class DashboardResponse(
    val statistiche_mezzi: List<StatisticaMezzo>,
    val statistiche_feedback: List<StatisticaFeedback>,
    val poi_piu_attivi: List<StatisticaPOI>
)